test/ApplicationSamplerTest.cpp

```diff
@@ -294,9 +302,9 @@ TEST_F(ApplicationSamplerTest, string_conversion)
 
 TEST_F(ApplicationSamplerTest, process_mapping)
 {
-    std::vector<int> expected { 4, 5, 77, 32 };
-    EXPECT_CALL(*m_mock_profile_sampler, cpu_rank())
-        .WillOnce(Return(expected));
+    std::vector<int> expected {42, 42};
+    EXPECT_CALL(*m_mock_status, get_process(_))
+        .WillRepeatedly(Return(42));
```

Comment:
if you want to test different values here, you can chain together multiple ` .WillOnce(Return(..))`